crates/bevy_color/src/linear_rgba.rs

```diff
@@ -0,0 +1,257 @@
+        let adjustment = luminance / current_luminance;
+        Self {
+            red: (self.red * adjustment).clamp(0., 1.),
+            green: self.green * adjustment.clamp(0., 1.),
+            blue: self.blue * adjustment.clamp(0., 1.),
+            alpha: self.alpha,
+        }
+    }
+
+    #[inline]
+    fn darker(&self, amount: f32) -> Self {
+        let mut result = *self;
+        result.adjust_lightness(-amount);
+        result
+    }
+
+    #[inline]
+    fn lighter(&self, amount: f32) -> Self {
+        let mut result = *self;
+        result.adjust_lightness(amount);
+        result
+    }
+}
+
+impl Mix for LinearRgba {
+    #[inline]
+    fn mix(&self, other: &Self, factor: f32) -> Self {
+        let n_factor = 1.0 - factor;
+        Self {
+            red: self.red * n_factor + other.red * factor,
+            green: self.green * n_factor + other.green * factor,
+            blue: self.blue * n_factor + other.blue * factor,
+            alpha: self.alpha * n_factor + other.alpha * factor,
+        }
+    }
+}
+
+impl Alpha for LinearRgba {
+    #[inline]
+    fn with_alpha(&self, alpha: f32) -> Self {
+        Self { alpha, ..*self }
+    }
+}
+
+impl EuclideanDistance for LinearRgba {
+    #[inline]
+    fn distance_squared(&self, other: &Self) -> f32 {
+        let dr = self.red - other.red;
+        let dg = self.green - other.green;
+        let db = self.blue - other.blue;
+        dr * dr + dg * dg + db * db
+    }
+}
+
+impl From<Srgba> for LinearRgba {
+    #[inline]
+    fn from(value: Srgba) -> Self {
+        Self {
+            red: value.red.nonlinear_to_linear_srgb(),
+            green: value.green.nonlinear_to_linear_srgb(),
+            blue: value.blue.nonlinear_to_linear_srgb(),
+            alpha: value.alpha,
+        }
+    }
+}
+
+impl From<LinearRgba> for Color {
+    fn from(value: LinearRgba) -> Self {
+        Color::RgbaLinear {
+            red: value.red,
+            green: value.green,
+            blue: value.blue,
+            alpha: value.alpha,
+        }
+    }
+}
+
+impl From<Color> for LinearRgba {
+    fn from(value: Color) -> Self {
+        match value.as_rgba_linear() {
+            Color::RgbaLinear {
+                red,
+                green,
+                blue,
+                alpha,
+            } => LinearRgba::new(red, green, blue, alpha),
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl From<LinearRgba> for [f32; 4] {
+    fn from(color: LinearRgba) -> Self {
+        [color.red, color.green, color.blue, color.alpha]
+    }
+}
+
+impl From<LinearRgba> for Vec4 {
+    fn from(color: LinearRgba) -> Self {
+        Vec4::new(color.red, color.green, color.blue, color.alpha)
+    }
+}
+
+#[allow(clippy::excessive_precision)]
+impl From<Oklaba> for LinearRgba {
+    fn from(value: Oklaba) -> Self {
+        let Oklaba { l, a, b, alpha } = value;
+
+        // From https://github.com/Ogeon/palette
```

Comment:
```suggestion
        // From https://github.com/Ogeon/palette/blob/e75eab2fb21af579353f51f6229a510d0d50a311/palette/src/oklab.rs#L312-L332
```

I'm not a copyright expert by any means, but do we need to attribute any credit to `palette` for this? E.g., in the `README.md`